core/src/main/java/org/mule/runtime/core/internal/processor/DelegateTransaction.java

```diff
@@ -216,5 +223,15 @@ public void resume() throws TransactionException {}
     public String getId() {
       return null;
     }
+
+    @Override
+    public Optional<ComponentLocation> getComponentLocation() {
+      return empty();
+    }
+
+    @Override
+    public void setComponentLocation(ComponentLocation componentLocation) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. ([squid:S1186](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1186))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7640&issues=AWilgAaSotIXnmsUpHDS&open=AWilgAaSotIXnmsUpHDS)

[ISSUE_KEY]: AWilgAaSotIXnmsUpHDS